fe/src/main/java/org/apache/doris/analysis/QueryStmt.java

```diff
@@ -419,6 +418,14 @@ public long getOffset() {
         return limitElement.getOffset();
     }
 
+    public void setAssertNumRowsElement(int desiredNumOfRows) {
+        this.assertNumRowsElement = new AssertNumRowsElement(desiredNumOfRows, toSql());
```

Comment:
toSql() is error-prone, maybe cause problem in some corner case. It will be better if there is original statement. Maybe you can leave a TODO here.